src/Backups/BackupsWorker.cpp

```diff
@@ -183,6 +173,17 @@ OperationID BackupsWorker::startMakingBackup(const ASTPtr & query, const Context
     String backup_name_for_logging = backup_info.toStringForLogging();
     try
     {
+        /// Check if there are no concurrent backups
+        if (num_active_backups && !allow_concurrent_backups)
+        {
+            /// If its an internal backup and we currently have 1 active backup, it could be the original query, validate using backup_uuid
+            if (!(num_active_backups == 1 && backup_settings.internal && getAllActiveBackupInfos().at(0).id == toString(*backup_settings.backup_uuid)))
+            {
+                addInfo(backup_id, backup_name_for_logging, backup_settings.internal, BackupStatus::BACKUP_FAILED);
```

Comment:
I think a better place for those checks would be after the call

```
addInfo(backup_id, backup_name_for_logging, backup_settings.internal, BackupStatus::CREATING_BACKUP);
```

below. Because if you move your check there all you will have to do is to throw an exception without this extra `addInfo()` call.

Reply:
If I moved the check after `addInfo`, then this condition cannot be checked so easily `getAllActiveBackupInfos().at(0).id == toString(*backup_settings.backup_uuid)` as `infos` is an `std::unordered_map`. Would have to check both the ids and match if one of it is current id and use the other one for backup_uuid verification. 
(also num_active_backups would be greater than 1, so checks will be not intuitive to read, but this is just for readability)
So to keep it simple, I thought it's better to keep it before adding the entry.

Reply:
You can check before `addInfo()`, but throw an exception later. Or both check and throw inside `addInfo()`.

Reply:
I think checking before addInfo() and throwing later is better.

Reply:
I don't think it's a good idea to separate checking & throwing exception. It's possible someone might add code in between. do you see any concerns with directly calling addInfo() with FAILED status ?

Reply:
added comment as discussed in below comment